b2share/modules/records/views.py

```diff
@@ -273,3 +302,82 @@ def post(self, **kwargs):
         location = url_for(endpoint, pid_value=pid.pid_value, _external=True)
         response.headers.extend(dict(location=location))
         return response
+
+class RecordsAbuseResource(RecordResource):
+
+    def post(self, **kwargs):
+        for v in ['abusecontent','message','email','copyright','zipcode',\
+                  'phone','illegalcontent','city','noresearch','name',\
+                  'affiliation','address','country']:
+            if v not in request.json:
+                msg = v + ' is required'
+                return jsonify({'Error':msg})
+
+        count = 0
+        for ii in ['noresearch', 'abusecontent', 'copyright', 'illegalcontent']:
+            if request.json[ii]: count += 1
+        if count != 1:
+            return jsonify({'Error':'From \'noresearch\', \'abusecontent\', \'copyright\', \'illegalcontent\' (only) one should be True'})
+
+        FriendlyReason = { 'abusecontent': 'Abuse or Inappropriate content',
+                            'copyright': 'Copyrighted material',
+                            'noresearch': 'No research data',
+                            'illegalcontent': 'Illegal content' }
+        msg_content = """
+            We have received new abuse report!
+            Link: """ + re.sub('/abuserecords\?$','',request.full_path) + """
+            Subject: " Abuse Report for a Record "
+            Reason: """ + [ FriendlyReason[ii] for ii in ['noresearch', 'abusecontent', 'copyright', 'illegalcontent'] if request.json[ii] ][0] + """
+            Message: """ + str(request.json['message']) + """
+            Full Name: """ + str(request.json['name']) + """
+            Affiliation: """ + str(request.json['affiliation']) + """
+            Email: """ + str(request.json['email']) + """
+            Address: """ + str(request.json['address']) + """
+            City: """ + str(request.json['city']) + """
+            Country: """ + str(request.json['country']) + """
+            Postal Code: """ + str(request.json['zipcode']) + """
+            Phone: """ + str(request.json['phone']) + """
+            """
+        support = str(current_app.config.get('SUPPORT_EMAIL'))
+        msg = Message("Abuse Report for a Record", sender=str(request.json['email']),
+                         recipients=[support], body=msg_content)
+        app = Flask('abuseapp')
+        app.config.update(MAIL_SUPPRESS_SEND=True)
+        InvenioMail(app)
+        with app.app_context():
+            app.extensions['mail'].send(msg)
```

Comment:
why not `current_app.extensions['mail'].send(msg)`? Why create a new flask app just for that? If it's really needed then it must be documented.